src/triage/component/postmodeling/contrast/group_evaluator.py

```diff
@@ -0,0 +1,325 @@
+        self.model_metrics = None
+
+    def __str__(self):
+        s = 'Model collection object for model_ids: {}'.format(self.model_id)
+        s += '\n Model Groups: {}'.format(self.model_group_id)
+        s += '\n Model types: {}'.format(self.model_type)
+        s += '\n Model hyperparameters: {}'.format(self.hyperparameters)
+        s += '\Matrix hashes (train,test): {}'.format((self.train_matrix_uuid,
+                                                      self.test_matrix_uuid))
+        return s
+
+    def _load_predictions(self):
+        preds = pd.read_sql("""
+                SELECT model_id,
+                       entity_id,
+                       as_of_date,
+                       score,
+                       label_value,
+                       COALESCE(rank_abs, RANK() 
+                       OVER(PARTITION BY model_id ORDER BY score DESC)) AS rank_abs,
+                       rank_pct,
+                       test_label_timespan
+                FROM test_results.predictions
+                WHERE model_id IN {}
+                AND label_value IS NOT NULL
+                """.format(tuple(self.model_id)),
+                con=db_engine)
+        self.preds = preds
+
+    def _load_feature_importances(self):
+        features = pd.read_sql("""
+               SELECT model_id,
+                      feature,
+                      feature_importance,
+                      rank_abs
+               FROM train_results.feature_importances 
+               WHERE model_id IN {}
+               """.format(tuple(self.model_id)),
+               con=db_engine)
+        self.feature_importances = features
+
+    def _load_metrics(self):
+        model_metrics = pd.read_sql("""
+                SELECT model_id,
+                       metric,
+                       parameter,
+                       value,
+                       num_labeled_examples,
+                       num_labeled_above_threshold
+                       num_positive_labels
+                FROM test_results.evaluations
+                WHERE model_id IN {}
+                """.format(tuple(self.model_id)),
+                con=db_engine)
+        self.model_metrics = model_metrics
+
+    def _rank_corr_df(self,
+                      model_pair,
+                      top_n,
+                      type):
+        '''
+        Calculates ranked correlations for ranked observations and features
+        using the stats.spearmanr scipy module.
+        Arguments: 
+            - model_pair (tuple): tuple with model_ids
+              observations or features
+            - top_n (int): number of rows to rank (top-k model)
+        '''
+
+        if type not in ['predictions', 'features']:
+            raise Exception('Wrong type! Rank correlation is not available'
+                    + '\n for {}. Try the following options:'.format(type) 
+                    + '\n predictions and features')
+
+        if type is 'predictions':
```

Comment:
Though this does work in C-Python, because the interpreter refuses to needlessly construct extra instances of immutable objects like strings, it's generally frowned upon, because you're treating the input as though it _must_ be the **same instance** as the one you're constructing dynamically here. Instead, you can just check `==` equality.

Reply:
Also ,this seems to be three different methods, you should split this function (remember, functions should have a single purpose).